docs/data/date-pickers/experimentation/CustomField.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import TextField from '@mui/material/TextField';
+import {
+  BaseSingleInputFieldProps,
+  DateValidationError,
+  FieldSection,
+} from '@mui/x-date-pickers/models';
+import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
+import { UseDateFieldProps } from '@mui/x-date-pickers/DateField';
+import {
+  BaseDateValidationProps,
+  DefaultizedProps,
+  splitFieldInternalAndForwardedProps,
+} from '@mui/x-date-pickers/internals';
+
+interface ReadOnlyDateFieldProps
+  extends DefaultizedProps<
```

Comment:
We have to use `DefaultizedProps` here because the validator expects a defaultized version of the validation props (with `minDate` and `maxDate` always populated for instance).

Two alternatives would be:

1. To support null values for those props in the validator (not great because right now the typing ensures that we never forget to pass a defaultized version on our codebase)
2. To export a hook to defaultized those props (that would be re-used by both the date picker and the date field), this would probably be the cleanest option